helix-lsp/src/client.rs

```diff
@@ -25,6 +26,12 @@ pub struct Client {
     offset_encoding: OffsetEncoding,
 }
 
+impl Drop for Client {
+    fn drop(&mut self) {
+        let _ = crate::block_on(self.force_shutdown());
```

Comment:
This needs a timeout or we can block far too long. I also think the behavior on drop is too implicit. This should happen as a part of application shutdown, here: https://github.com/helix-editor/helix/blob/8664d70e731c73fa34dda293b4f6b6dec80a3273/helix-term/src/application.rs#L333

We can use
```rust
futures::future::join_all(language_servers.into_iter().map(|client| client.force_shutdown())).await
```
to shut down all language servers concurrently.